### ISSUE TITLE
Text editor missing horizontal scrolling or line wrapping

### ISSUE THREAD

**Issue by TeamDman** [TeamDman/SuperFactoryManager#546]
Lines of text that exceed the width of the terminal are in a poor state with regards to user experience.
The vanilla component had line wrapping behaviour that has been overridden as an implementation detail of how syntax highlighting currently works.
A horizontal scrollbar sounds trivial, but the text editor and scroll widget are currently vanilla components that change between versions and historically have been a source of bugs.
